### ISSUE TITLE
Unable to parse timestamp of type millisecond since unix epoch

### ISSUE THREAD

**Issue by jeromekleinen-kbc** [timberio/vector#7360]
### Vector Version

```
vector 0.13.1 (v0.13.1 x86_64-unknown-linux-gnu 2021-04-29)
```

### Vector Configuration File

```
$ parse_timestamp!("1620291751177", "%s%3f")
function call error for "parse_timestamp" at (0:42): Invalid timestamp "1620291751177": premature end of input

$ parse_timestamp!("1620291751.177", "%s%.3f")
t'2021-05-06T09:02:31.177Z'
```

### Debug Output

N/A

### Expected Behavior

The first parse_timestamp command should have been able to parse the timestamp.

### Actual Behavior

It didn't parse the timestamp.

**Comment by jszwedko** [comment]
I dug into this a little and couldn't find a format specifier that would work for unix milliseconds in the time parsing library we use, chrono. I opened: https://github.com/chronotope/chrono/issues/560

**Comment by yalinglee** [comment]
Would the solution proposed in the chrono issue also work for parsing unix microseconds like `1638777946995336`?

**Comment by jszwedko** [comment]
@yalinglee yes, it looks like it would work for millisecond, microsecond, and nanosecond timestamps.

https://github.com/chronotope/chrono/pull/561/files#diff-abd7a03adfeaa1c9ca4623371f789cf48c5d794d82457a4068ad5be00fe7c784R225-R227

**Comment by yalinglee** [comment]
@jszwedko That's great to hear! Any chance this might be addressed in the near future? We'd love to try out vector but this is currently a blocker.

**Comment by jszwedko** [comment]
Unfortunately we are blocked on `chrono` until the merge that. I did leave a comment though so hopefully it'll get some attention soon.

As a workaround, you can use arithmetic to convert the timestamp to a float timestamp which Vector will interpret as seconds (with fractional seconds):

```
timestamp = 1620291751177
to_timestamp(timestamp / 1000000)
```

Does that make sense?

**Comment by yalinglee** [comment]
@jszwedko Thank you so much for getting the ball rolling on this, and yes the workaround should work in the meantime!
